src/gl_fns.rs

```diff
@@ -15,9 +15,10 @@ impl GlFns
 {
     pub unsafe fn load_with<'a, F>(loadfn: F) -> Rc<Gl> where F: FnMut(&str) -> *const c_void {
         let ffi_gl_ = GlFfi::load_with(loadfn);
-        Rc::new(GlFns {
+        let fns = GlFns {
             ffi_gl_: ffi_gl_,
-        }) as Rc<Gl>
+        };
+        ErrorCheckingGl::maybe_wrap(fns)
```

Comment:
What's most bothering me about the PR is that the switch to use asserting wrapper can really be a run-time switch (as opposed to compile time feature). Can't we just expose `ErrorCheckingGl` to the user and let them call `wrap` if they want to?

Reply:
That's fair.